src/tt21100.toit

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the package's LICENSE file.
+
+import gpio
+import i2c
+
+TT2100_FLIP_X  := 0x1
+TT2100_FLIP_Y  := 0x2
+TT2100_FLIP_XY := 0x4
+
+INVALID_VALUE_ ::= -1
+
+bytes2ushort bytes/ByteArray -> int:
+  return (bytes[0] + (bytes[1] << 8))
+
+class Action:
+  TOUCH_PAD ::= 0x1
+  PRESS_BUTTON ::= 0x3
+
+  type/int
+  time_stamp/int
+
+  x/int := INVALID_VALUE_
+  y/int := INVALID_VALUE_
+  touch_pressure/int := INVALID_VALUE_
+
+  button/int := INVALID_VALUE_
+  button_pressure/int := INVALID_VALUE_
+
+  constructor data/ByteArray:
+    length/int := bytes2ushort data[0..2]
+    type = data[2]
+    time_stamp = bytes2ushort data[3..5]
+
+    if type == TOUCH_PAD:
+      num/int := data[5] & 0x1f
+      if num > 0:
+        x = bytes2ushort data[9..11]
+        y = bytes2ushort data[11..13]
+        touch_pressure = data[13]
+    else if type == PRESS_BUTTON:
+      if data[5] & 0x1:
+        button = 1
+      else if data[5] & 0x2:
+        button = 2
+      else if data[5] & 0x4:
+        button = 3
+      
+      button_pressure = bytes2ushort data[6..8]
+
+  /**
+  string format:
+      action name(timestamp): action details description
+  */
+  stringify -> string:
+    if type == TOUCH_PAD:
+      if x != INVALID_VALUE_:
+        return "touch-pad($time_stamp): X=$x Y=$y Pressure=$touch_pressure"
+      else:
+        return "touch-pad($time_stamp): done"
+    else if type == PRESS_BUTTON:
+      return "presss-button($time_stamp): Button=$button Pressure=$button_pressure"
+    else:
+      return "Action is not supported"
+
+class Driver:
```

Comment:
I would rename to `Tt21100`.
```suggestion
class Tt21100:
```

That said: we haven't applied this consistently. Sometimes we used `Driver` and sometimes we used the name of the sensor. Up to you.

Reply:
Yeah, in my opinion I would like also use `Tt21100`, the `Driver` is from `m5stack2` project.